invenio_oauthclient/contrib/cern.py

```diff
@@ -209,16 +211,68 @@ def account_setup(remote, token):
 
     response = remote.get(REMOTE_APP_RESOURCE_API_URL)
     user = token.remote_account.user
+    if not user:
+        raise Exception('`user` should not be None, something bad happened!')
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+
+        # update session object
+        groups_old = current_user.info['group']
+        groups_new = fetch_groups(res['Group'])
+        current_user.info['group'] = groups_new
```

Comment:
@ludmilamarian, correct. Upon login time, external e-groups are synchronized into the usergroup table, then any usergroup (external or internal) from the usergroup table is loaded into `user_info` and it's then used for checking authorizations of that given user.

There is also a last use-case though:
https://github.com/inveniosoftware/invenio/blob/legacy/modules/webaccess/lib/access_control_firerole.py#L174

where the function `acc_firerole_extract_emails()` has a special `if CFG_CERN_SITE` to actually use external e-groups from the usergroup table, which is then used by `acc_get_authorized_emails()`. Which finally is only used in the infamous `publiline.py` monster, and in `webcomment`. In order to discover who should really be notified to moderate comments (even considering fireroles)